cpp/environment/AngularSeparation.h

```diff
@@ -5,57 +5,83 @@
 #define ANGULAR_SEPARATION_H
 
 #include "NeighborList.h"
+#include "NeighborQuery.h"
 #include "VectorMath.h"
 
 /*! \file AngularSeparation.h
-    \brief Compute the angular separation for each particle.
+    \brief Compute the angular separations.
 */
 
 namespace freud { namespace environment {
 
-float computeSeparationAngle(const quat<float> ref_q, const quat<float> q);
+//! Compute the angular separation for a set of points
+/*! Given a set of global orientations, this method accepts a set of
+ * orientations that are compared against the global orientations to determine
+ * the total angular distance between them. The output is an array of shape
+ * (num_orientations, num_global_orientations) containing the pairwise
+ * separation angles between the provided orientations and global orientations.
+ */
+class AngularSeparationGlobal
+{
+public:
+    //! Constructor
+    AngularSeparationGlobal() {}
 
-float computeMinSeparationAngle(const quat<float> ref_q, const quat<float> q, const quat<float>* equiv_qs,
-                                unsigned int Nequiv);
+    //! Destructor
+    ~AngularSeparationGlobal() {}
 
-//! Compute the angular separation for a set of points
-/*!
+    //! Compute the angular separation with respect to global orientation
+    void compute(const quat<float>* global_orientations, unsigned int n_global,
+                 const quat<float>* orientations, unsigned int n_points,
+                 const quat<float>* equiv_orientations, unsigned int n_equiv_orientations);
+
+    //! Returns the last computed global angle array
+    const util::ManagedArray<float> &getAngles() const
+    {
+        return m_angles;
+    }
+
+protected:
```

Comment:
I think this should be private for consistency with other classes, unless there's a benefit to `protected` that I'm not aware of.
```suggestion
private:
```

Reply:
There's not really a reason to prefer one over the other for classes with no children. I changed it because I was originally going to have a common parent then realized there was no logic in common. I can change it back to be consistent.